### PR TITLE
Add acceptance tests and docs for helm repository and helm charts data source

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Add acceptance tests and docs for helm repository and helm charts data source

Acceptance test output
Screenshot 2023-09-27 at 7 51 52 AM

Screenshot 2023-09-27 at 7 52 33 AM

-----------------------------------------------------------

Mock Accep. tests output
Screenshot 2023-09-27 at 7 53 18 AM

Screenshot 2023-09-27 at 7 53 47 AM

### CODE REVIEW

File: docs/data-sources/helm_charts.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Title: "Helm Charts Data Source"
+Description: |-
+    Read the Helm Charts from TMC.
+---
+
+# Helm Charts Repository
+
+This resource allows you to get list of Helm Charts through Tanzu Mission Control.
+
+The Available tab on the Catalog page in the Tanzu Mission Control console shows the Helm charts that are available.
```
Comment (ishangupta-ds) [inline docs/data-sources/helm_charts.md:11]: `Available` tab is correct?

File: internal/resources/helmcharts/helm_charts_data_spurce_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+	testhelper "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/testing"
+)
+
+func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
+	return &testAcceptanceConfig{
+		Provider:            initTestProvider(t),
+		ChartResource:       chartResource,
+		ChartDataSourceVar:  chartDataSourceVar,
+		ChartMetadataName:   "zookeeper",
```
Comment (ishangupta-ds) [inline internal/resources/helmcharts/helm_charts_data_spurce_test.go:25]: Could have been a constant.

File: internal/resources/helmcharts/helm_charts_data_spurce_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
+/*
```
Comment (ishangupta-ds) [inline internal/resources/helmcharts/helm_charts_data_spurce_test.go:1]: Fix the filename. Check all other files for Helm.

File: templates/data-sources/helm_charts.md.tmpl [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+Title: "Helm Charts Data Source"
+Description: |-
+    Read the Helm Charts from TMC.
+---
+
+# Helm Charts Repository
```
Comment (ishangupta-ds) [inline templates/data-sources/helm_charts.md.tmpl:7]: Shouldn't this be Helm Repository Charts?

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: templates/data-sources/helm_charts.md.tmpl (hunk 4 above)
Comment (Shubbhang351) [reply templates/data-sources/helm_charts.md.tmpl]: Updated PTAL

File: internal/resources/helmcharts/helm_charts_data_spurce_test.go (hunk 3 above)
Comment (Shubbhang351) [reply internal/resources/helmcharts/helm_charts_data_spurce_test.go]: Updated PTAL

File: internal/resources/helmcharts/helm_charts_data_spurce_test.go (hunk 2 above)
Comment (Shubbhang351) [reply internal/resources/helmcharts/helm_charts_data_spurce_test.go]: Updated PTAL

File: docs/data-sources/helm_charts.md (hunk 1 above)
Comment (Shubbhang351) [reply docs/data-sources/helm_charts.md]: Updated PTAL

**Review by ishangupta-ds — APPROVED** [review]

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**